src/tests/sqlsmith/src/sqlreduce/passes/pullup.rs

```diff
@@ -0,0 +1,293 @@
+        "case_pullup".to_owned()
+    }
+
+    fn get_reduction_points(&self, ast: Ast) -> Vec<usize> {
+        let mut reduction_points = Vec::new();
+        if let Statement::Query(query) = &ast
+            && let SetExpr::Select(select) = &query.body
+        {
+            for (i, item) in select.projection.iter().enumerate() {
+                if let SelectItem::UnnamedExpr(expr) = item
+                    && let Expr::Case { .. } = expr
+                {
+                    reduction_points.push(i);
+                }
+            }
+        }
+        reduction_points
+    }
+
+    fn apply_on(&self, ast: &mut Ast, reduction_points: Vec<usize>) -> Ast {
+        if let Statement::Query(query) = ast
+            && let SetExpr::Select(select) = &mut query.body
+        {
+            for i in reduction_points {
+                if let SelectItem::UnnamedExpr(ref mut expr) = select.projection[i]
+                    && let Expr::Case { results, .. } = expr
+                {
+                    *expr = results[0].clone();
+                }
+            }
+        }
+
+        ast.clone()
+    }
+}
+
+/// Simplify `ROW` expressions in SELECT projections by pulling up the first element.
+///
+/// This transformation replaces a `ROW(...)` expression with its first component,
+/// reducing the structure without preserving full semantics. It is intended for
+/// query reduction purposes where correctness (e.g., error reproduction) remains.
+///
+/// Example:
+/// ```sql
+/// SELECT ROW(1, 2, 3);
+/// ```
+/// Will be reduced to:
+/// ```sql
+/// SELECT 1;
+/// ```
+pub struct RowPullup;
+
+impl Transform for RowPullup {
+    fn name(&self) -> String {
+        "row_pullup".to_owned()
+    }
+
+    fn get_reduction_points(&self, ast: Ast) -> Vec<usize> {
+        let mut reduction_points = Vec::new();
+        if let Statement::Query(query) = &ast
+            && let SetExpr::Select(select) = &query.body
+        {
+            for (i, item) in select.projection.iter().enumerate() {
+                if let SelectItem::UnnamedExpr(expr) = item
+                    && let Expr::Row { .. } = expr
+                {
+                    reduction_points.push(i);
+                }
+            }
+        }
+        reduction_points
+    }
+
+    fn apply_on(&self, ast: &mut Ast, reduction_points: Vec<usize>) -> Ast {
+        if let Statement::Query(query) = ast
+            && let SetExpr::Select(select) = &mut query.body
+        {
+            for i in reduction_points {
+                if let SelectItem::UnnamedExpr(ref mut expr) = select.projection[i]
+                    && let Expr::Row(elements) = expr
+                {
+                    *expr = elements[0].clone();
+                }
+            }
+        }
+        ast.clone()
+    }
+}
+
+/// Simplify array constructors in SELECT projections by pulling up the first element.
+///
+/// This transformation replaces an `ARRAY[...]` expression with its first element.
```

Comment:
Not sure how effective this will be, since it changes the type of the select item, invalidating references which depend on the array type. Do you mean to do `SELECT ARRAY[1]` instead in the example?